pkg/driver/controller_test.go

```diff
@@ -3112,450 +3114,251 @@ func TestControllerPublishVolume(t *testing.T) {
-				req := &csi.ControllerUnpublishVolumeRequest{}
-
-				ctx := context.Background()
-
-				mockCtl := gomock.NewController(t)
-				defer mockCtl.Finish()
-
-				mockCloud := cloud.NewMockCloud(mockCtl)
-
-				awsDriver := controllerService{
-					cloud:         mockCloud,
-					inFlight:      internal.NewInFlight(),
-					driverOptions: &DriverOptions{},
-				}
-
-				if _, err := awsDriver.ControllerUnpublishVolume(ctx, req); err != nil {
-					srvErr, ok := status.FromError(err)
-					if !ok {
-						t.Fatalf("Could not get error status code from error: %v", srvErr)
-					}
-					if srvErr.Code() != codes.InvalidArgument {
-						t.Fatalf("Expected error code %d, got %d message %s", codes.InvalidArgument, srvErr.Code(), srvErr.Message())
-					}
-				} else {
-					t.Fatalf("Expected error %v, got no error", codes.InvalidArgument)
-				}
+			name:      "Return success when volume not found during DetachDisk operation",
+			volumeId:  "vol-not-found",
+			nodeId:    expInstanceID,
+			errorCode: codes.OK,
+			mockDetach: func(mockCloud *cloud.MockCloud, ctx context.Context, volumeId string, nodeId string) {
+				mockCloud.EXPECT().DetachDisk(gomock.Eq(ctx), volumeId, nodeId).Return(cloud.ErrNotFound)
 			},
+			expResp: &csi.ControllerUnpublishVolumeResponse{},
 		},
 		{
-			name: "fail no NodeId",
-			testFunc: func(t *testing.T) {
-				req := &csi.ControllerUnpublishVolumeRequest{
-					VolumeId: "vol-test",
-				}
+			name:      "Invalid argument error when no VolumeId provided",
+			volumeId:  "",
+			nodeId:    expInstanceID,
+			errorCode: codes.InvalidArgument,
+		},
+		{
+			name:      "Invalid argument error when no NodeId provided",
+			volumeId:  "vol-test",
+			nodeId:    "",
+			errorCode: codes.InvalidArgument,
+		},
+		{
+			name:      "Internal error when DetachDisk operation fails",
+			volumeId:  "vol-test",
+			nodeId:    expInstanceID,
+			errorCode: codes.Internal,
+			mockDetach: func(mockCloud *cloud.MockCloud, ctx context.Context, volumeId string, nodeId string) {
+				mockCloud.EXPECT().DetachDisk(gomock.Eq(ctx), volumeId, nodeId).Return(errors.New("test error"))
+			},
+		},
+		{
+			name:      "Aborted error when operation already in-flight",
+			volumeId:  "vol-test",
+			nodeId:    expInstanceID,
+			errorCode: codes.Aborted,
+		},
+	}
 
-				ctx := context.Background()
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			req := &csi.ControllerUnpublishVolumeRequest{
+				NodeId:   tc.nodeId,
+				VolumeId: tc.volumeId,
+			}
 
-				mockCtl := gomock.NewController(t)
-				defer mockCtl.Finish()
+			ctx := context.Background()
 
-				mockCloud := cloud.NewMockCloud(mockCtl)
+			awsDriver, mockCtl, mockCloud := createControllerService(t)
+			defer mockCtl.Finish()
 
-				awsDriver := controllerService{
-					cloud:         mockCloud,
-					inFlight:      internal.NewInFlight(),
-					driverOptions: &DriverOptions{},
-				}
+			if tc.name == "Aborted error when operation already in-flight" {
```

Comment:
Never do this, needs to be changed. Add a setup function or similar to the test parameters, do not hardcode behavior based on test name.

Changing the test name should NEVER affect the test's behavior.

Reply:
Done.